languages/go/exercises/concept/maps/.docs/instructions.md

```diff
@@ -0,0 +1,80 @@
+A friend of you has an old wholesale store called **Gross Store**, the name came from the quantity of the item that the store sell, it's all in [gross unit][gross-unit]. Your friend asked you to implement a point of sale (POS) system for his store, **but first, you want to build a prototype for it, in your prototype, your system will only record the quantity**. Your friend gave you a list of measurements to help you:
+
+| Unit               | Score |
+| ------------------ | ----- |
+| quarter_of_a_dozen | 3     |
+| half_of_a_dozen    | 6     |
+| dozen              | 12    |
+| small_gross        | 120   |
+| gross              | 144   |
+| great_gross        | 1728  |
+
+## 1. Store the unit of measurement in your program
+
+In order to use the measurement, you need to store the measurement in your program.
```

Comment:
I'm thinking if there was a way in which we could check if students actually defined the map correctly. You used to have an empty variable in the stub implementation file. I quite liked that actually, as you could then have a unit test verify its contents (and the code would compile). The main benefit of this is that we force the student to define the mapping as a map, instead of possibly defining it as a function with if/else statements. Furthermore, as students will need to define a non-empty map, which means they must know how to do that (they don't necessarily know that now, do they?)

Reply:
> You used to have an empty variable in the stub implementation file. I quite liked that actually, as you could then have a unit test verify its contents

Well, the stub was commented and untyped, it was more like hint rather than a stub. If the student uncomment the stub without adding any value, it won't compile.

> I'm thinking if there was a way in which we could check if students actually defined the map correctly


To my knowledge, since go is compiled, there's no way to check if the map implemented in global map or function in runtime. Instead, what we can do is to have an ast check, which I assume the representer would cover, right? (I still doesn't understand the concept of analyzer and representer)

> Furthermore, as students will need to define a non-empty map, which means they must know how to do that (they don't necessarily know that now, do they?)

Ah, good point. the `introduction.md` doesn't cover map literal. I think I just have to map literal in `introduction.md`??

By the way, I readded the hint in 0aa58c0 , nice catch @ErikSchierboom

Reply:
What if you'd add a function to return the units?

```go
func Units() map[string]int
```

That could work, right?

Reply:
Yes that should work, and I also can pass the units map though the function arguments.

So the function would take  4 parameters, don't you think thee parameters are a bit too much?

Just want to let you know, I'm currently implementing your suggestion.

Reply:
Ok did it in, 7ee66cc 

I wasn't expect creating an exercise is this hard, there're lots of things to consider and the style ... IDK how to phrase this but I sort of how to change style more towards the exercise rather than implement it in my own style, that's pretty much new to me.

Reply:
By the way, thanks for your patience and review Erik, really appreciate it

Reply:
> IDK how to phrase this but I sort of have to change style more towards the exercise rather than implement it in my own style, that's pretty much new to me.

Yes, I can empathize with this, as I've had the same experience when building C# and F# exercises. It is sometimes really hard to design an exercise such that the student is guided towards a correct solution and one that we can then verify through unit tests.

Reply:
> Yes that should work, and I also can pass the units map though the function arguments.
> 
> So the function would take 4 parameters, don't you think thee parameters are a bit too much?

I agree that 4 parameters is a bit much. What if you didn't pass the units as a parameter? Having the student define the units as the first task will likely mean that they'll be re-using that function in their subsequent functions, without needing to pass it in as a parameter. What do you think?

Reply:
> Having the student define the units as the first task will likely mean that they'll be re-using that function in their subsequent functions, without needing to pass it in as a parameter.

I think I would prefer that.